sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6001,6 +6008,10 @@ pi_result piEnqueueMemBufferCopy(pi_queue Queue, pi_mem SrcBuffer,
                                  pi_event *Event) {
   PI_ASSERT(SrcBuffer && DstBuffer, PI_INVALID_MEM_OBJECT);
   PI_ASSERT(Queue, PI_INVALID_QUEUE);
+
+  std::scoped_lock Lock(Queue->PiQueueMutex, SrcBuffer->Mutex,
```

Comment:
The "std::scoped_lock" will lock all mutex for exclusive use, which looks like unnecessary many times.
Can we adopt a more fine-grain approach, like this:
```
    std::shared_lock L1(SrcBuffer->Mutex, std::defer_lock);
    std::unique_lock L2(DstBuffer->Mutex, std::defer_lock);
    std::unique_lock L3(Queue->Mutex, std::defer_lock);
    std::scoped_lock lockall(L1, L2, L3);
```
?